src/zenml/post_execution/artifact.py

```diff
@@ -74,3 +75,48 @@ def read(
         from zenml.utils.materializer_utils import load_artifact
 
         return load_artifact(self.model)
+
+    def visualize(self) -> None:
+        """Visualize the artifact in notebook environments.
+
+        Raises:
+            RuntimeError: If not in a notebook environment.
+        """
+        from IPython.core.display import HTML, Image, Markdown, display
+
+        from zenml.environment import Environment
+        from zenml.utils.materializer_utils import load_artifact_visualization
+
+        if not Environment.in_notebook() and not Environment.in_google_colab():
+            raise RuntimeError(
+                "The `output.visualize()` method is only available in Jupyter "
+                "notebooks. In all other runtime environments, please open "
+                "your ZenML dashboard using `zenml up` and view the "
+                "visualizations by clicking on the respective artifacts in the "
+                "pipeline run DAG instead."
+            )
+
+        if not self.model.visualizations:
+            return
+
+        display(Markdown(f"### {self.model.name}"))
+        for i in range(len(self.model.visualizations)):
+            visualization = load_artifact_visualization(self.model, index=i)
+            if visualization.type == VisualizationType.IMAGE:
+                display(Image(visualization.value))
+            elif visualization.type == VisualizationType.HTML:
+                display(HTML(visualization.value))
+            elif visualization.type == VisualizationType.MARKDOWN:
+                display(Markdown(visualization.value))
+            elif visualization.type == VisualizationType.CSV:
+                assert isinstance(visualization.value, str)
+                table = "<table>"
+                for row in visualization.value.splitlines():
+                    table += "<tr>"
+                    for cell in row.split(","):
+                        table += f"<td>{cell}</td>"
+                    table += "</tr>"
+                table += "</table>"
```

Comment:
Wondering also if this needs extracting, assuming visualizations will only expand and/or become more of a thing.

Reply:
Moved it to new file `utils/visualization_utils.py` 👍